### ISSUE TITLE
Uploading file onto S3

### ISSUE THREAD

**Issue by nudgetman** [crocodic-studio/crudbooster#650]
I'm at a blunder with S3. This is what I've got:

- the .env settings for S3 right
- changed my 'config/filesystem.php' file to use the 's3' driver
- confirmed that a folder with 'Y-m' format created in the bucket is created

Having all this, I still cannot upload the file onto the bucket.

Anybody else having the issue or able to fix this?

Thank you

**Comment by fherryfherry** [comment]
you may try to update the latest version

**Comment by nudgetman** [comment]
will give it and try and update here 👍

**Comment by nudgetman** [comment]
I've tested the upload with the latest version and still nothing in my S3 and the image goes to the usual place in the storage folder. Is there a way to see if I'm actually connecting anything to s3?

**Comment by nudgetman** [comment]
I've done another update & now it works! 😄 

However, there's another issue. The link is set to as if it in local and not on S3.

**Comment by dtannen** [comment]
yes-- need option to add STORAGE_URL.  Now it looks for local /storage.

**Comment by sameerpanjwani** [comment]
I found a solution where you change the URL being stored in the `hook_before_add` function, e.g.

```
public function hook_before_add(&$postdata) {        
	        //Your code here
            $postdata["product_image_url"]=\Storage::disk("s3")->url($postdata['product_image_url']);
	    }
```
